From 86a73ccfc534e08e2513fbc95420bbed9f53ce73 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E5=B0=8F=E5=A4=AA?= <nospam@kota.moe>
Date: Sat, 6 Apr 2024 08:02:14 +1100
Subject: [PATCH] notifications: Fix use-after-free in timeout_bubble()

gtk_widget_destroy(bubble) might destroy the last reference to the bubble,
making the write to timeout_id a potential use-after-free

Fixes https://gitlab.gnome.org/GNOME/gnome-flashback/-/issues/96
---
 gnome-flashback/libnotifications/gf-bubble.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gnome-flashback/libnotifications/gf-bubble.c b/gnome-flashback/libnotifications/gf-bubble.c
index eabb4fe..00cc4a1 100644
--- a/gnome-flashback/libnotifications/gf-bubble.c
+++ b/gnome-flashback/libnotifications/gf-bubble.c
@@ -190,10 +190,10 @@ timeout_bubble (gpointer user_data)
 
   bubble = GF_BUBBLE (user_data);
   priv = gf_bubble_get_instance_private (bubble);
+  priv->timeout_id = 0;
 
   gtk_widget_destroy (GTK_WIDGET (bubble));
 
-  priv->timeout_id = 0;
   return G_SOURCE_REMOVE;
 }
 
-- 
2.44.0

